js/bibox.js

```diff
@@ -557,47 +596,30 @@ module.exports = class bibox extends Exchange {
-        if ('error' in response) {
-            if ('code' in response['error']) {
-                let code = response['error']['code'];
-                if (code === '2033')
-                    // \u64cd\u4f5c\u5931\u8d25\uff01\u8ba2\u5355\u5df2\u5b8c\u6210\u6216\u5df2\u64a4\u9500
-                    // operation failed! Orders have been completed or revoked
-                    // e.g. trying to cancel a filled order
-                    throw new OrderNotFound (message);
-                else if (code === '2067')
-                    // https://github.com/ccxt/ccxt/issues/2338
-                    //  { "error": { "code": "2067", "msg": "暂不支持市价单"}, "cmd": "orderpending/trade" }
-                    // "Does not support market orders"
-                    throw new InvalidOrder (message);
-                else if (code === '2068')
-                    // \u4e0b\u5355\u6570\u91cf\u4e0d\u80fd\u4f4e\u4e8e
-                    // The number of orders can not be less than
-                    throw new InvalidOrder (message);
-                else if (code === '3012')
-                    throw new AuthenticationError (message); // invalid apiKey
-                else if (code === '3024')
-                    throw new PermissionDenied (message); // insufficient apiKey permissions
-                else if (code === '3025')
-                    throw new AuthenticationError (message); // signature failed
-                else if (code === '4000')
-                    // \u5f53\u524d\u7f51\u7edc\u8fde\u63a5\u4e0d\u7a33\u5b9a\uff0c\u8bf7\u7a0d\u5019\u91cd\u8bd5
-                    // The current network connection is unstable. Please try again later
-                    throw new ExchangeNotAvailable (message);
-                else if (code === '4003')
-                    throw new DDoSProtection (message); // server is busy, try again later
+    handleErrors (code, reason, url, method, headers, body) {
+        if (body.length > 0) {
+            if (body[0] === '{') {
+                let response = JSON.parse (body);
+                if ('error' in response) {
+                    if ('code' in response['error']) {
+                        let code = this.safeString (response['error'], 'code');
+                        let errorCodes = Object.keys (this.exceptions);
+                        for (let i = 0; i < errorCodes.length; i++) {
+                            if (code === errorCodes[i]) {
+                                throw new this.exceptions[errorCodes[i]] (this.id + ' ' + body);
+                            }
+                        }
+                    }
+                    throw new ExchangeError (this.id + ': "error" in response: ' + body);
+                }
+                if (!('result' in response))
+                    throw new ExchangeError (this.id + ' ' + body);
+                if (method === 'GET') {
+                    return response;
+                } else {
+                    return response['result'][0];
```

Comment:
Oh, now I see why it's broken... It never uses this return...

Reply:
No worries, will fix it shortly.

Reply:
Working in python now cheers

Reply:
Thanks, @frosty00 !)